lib/workos/directory_user.rb

```diff
@@ -44,6 +44,15 @@ def to_json(*)
       }
     end
 
+    def [](attribute_name)
+      # Warn the user not to use the Hash-style access for attributes.
+      warning_message = "WARNING: The Hash style access for DirectoryUser attributes is deprecated and will be removed
+in a future version. Please use user.#{attribute_name} instead of user['#{attribute_name}']"
+      puts warning_message
+
+      to_json[attribute_name.to_sym]
```

Comment:
Minor difference, but we could avoid allocating the full `to_json` hash for each property access by using `instance_variable_get`:

```suggestion
      instance_variable_get("@#{attribute_name}")
```